pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -145,8 +158,24 @@ func (meta *predicateMetadata) RemovePod(deletedPod *v1.Pod) error {
 	if deletedPodFullName == schedutil.GetPodFullName(meta.pod) {
 		return fmt.Errorf("deletedPod and meta.pod must not be the same")
 	}
-	// Delete any anti-affinity rule from the deletedPod.
-	delete(meta.matchingAntiAffinityTerms, deletedPodFullName)
+
+	// Delete pod from matching topology pairs map
+	for _, pair := range meta.antiAffinityPodToTopologyPairs[deletedPod] {
+		for index, pod := range meta.topologyPairToAntiAffinityPods[pair] {
+			if pod == deletedPod {
+				podsList := meta.topologyPairToAntiAffinityPods[pair]
+				podsList[index] = podsList[len(podsList)-1]
+				if len(podsList) <= 1 {
+					delete(meta.topologyPairToAntiAffinityPods, pair)
+				} else {
+					meta.topologyPairToAntiAffinityPods[pair] = podsList[:len(podsList)-1]
+				}
```

Comment:
add a `break` here to jump out of the loop early

Reply:
I still think we can `break` here as it already found the target "deletePod". @mohamed-mehany can you recheck?

Reply:
@Huang-Wei  That was already added in a subsequent commit(5d46098):

https://github.com/kubernetes/kubernetes/blob/5d46098a757005175a6e468c1b46ac6de2a6dbe1/pkg/scheduler/algorithm/predicates/metadata.go#L161-L170